components/org.wso2.carbon.identity.sts.passive.ui/src/main/java/org/wso2/carbon/identity/sts/passive/ui/PassiveSTS.java

```diff
@@ -345,22 +352,18 @@ private void persistWreply(ResponseToken responseToken, HttpSession session) {
     }
 
     private void sendToAuthenticationFramework(HttpServletRequest request, HttpServletResponse response,
-                                               String sessionDataKey, SessionDTO sessionDTO) throws IOException {
+                                               String sessionDataKey, SessionDTO sessionDTO)
+            throws IOException, PassiveSTSException {
 
         String commonAuthURL;
         try {
             commonAuthURL = ServiceURLBuilder.create().addPath(FrameworkConstants.COMMONAUTH).build()
                     .getAbsolutePublicURL();
         } catch (URLBuilderException e) {
-            throw new IOException("Error occurred while building the commonauth URL during login.", e);
+            throw new PassiveSTSException("Error occurred while building the commonauth URL during login.", e);
         }
 
-        String selfPath;
-        try {
-            selfPath = ServiceURLBuilder.create().addPath(request.getRequestURI()).build().getRelativeInternalURL();
-        } catch (URLBuilderException e) {
-            throw new IOException("Error occurred while building the commonauth caller path URL during login.", e);
-        }
+        String selfPath = request.getRequestURI();
```

Comment:
@Abilashini the suggestion was to use the exact path instead of request.getRequestURI() only. I think we can use the ServiceURLBuilder even in that case

So the code would be,

```
String selfPath;
        try {
            selfPath = ServiceURLBuilder.create().addPath("passivests").build().getRelativeInternalURL();
        } catch (URLBuilderException e) {
            throw new IOException("Error occurred while building the commonauth caller path URL during login.", e);
        }
```

"passivests" can be added via a constant too.

@malithie : is my understanding correct?

Reply:
I removed it to be addressed in a separate PR for tenant qualifying inbound URLs